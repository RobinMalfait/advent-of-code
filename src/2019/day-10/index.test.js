import { promises } from 'fs'
import { resolve } from 'path'

import part1 from './part-1.js'
import part2 from './part-2.js'

let data = promises.readFile(resolve(process.cwd(), 'data', '2019-10.txt'), 'utf8')

describe('Part 1', () => {
  it.each([
    [8, '3,4', ['.#..#', '.....', '#####', '....#', '...##'].join('\n')],
    [
      33,
      '5,8',
      ['......#.#.', '#..#.#....', '..#######.', '.#.#.###..', '.#..#.....', '..#....#.#', '#..#....#.', '.##.#..###', '##...#..#.', '.#....####'].join('\n'),
    ],
    [
      35,
      '1,2',
      ['#.#...#.#.', '.###....#.', '.#....#...', '##.#.#.#.#', '....#.#.#.', '.##..###.#', '..#...##..', '..##....##', '......#...', '.####.###.'].join('\n'),
    ],
    [
      41,
      '6,3',
      ['.#..#..###', '####.###.#', '....###.#.', '..###.##.#', '##.##.#.#.', '....###..#', '..#.#..#.#', '#..#.#.###', '.##...##.#', '.....#.#..'].join('\n'),
    ],
    [
      210,
      '11,13',
      [
        '.#..##.###...#######',
        '##.############..##.',
        '.#.######.########.#',
        '.###.#######.####.#.',
        '#####.##.#.##.###.##',
        '..#####..#.#########',
        '####################',
        '#.####....###.#.#.##',
        '##.#################',
        '#####.##.###..####..',
        '..######..##.#######',
        '####.##.####...##..#',
        '.#####..#.######.###',
        '##...#.##########...',
        '#.##########.#######',
        '.####.#.###.###.#.##',
        '....##.##.###..#####',
        '.#.#.###########.###',
        '#.#.#.#####.####.###',
        '###.##.####.##.#..##',
      ].join('\n'),
    ],
  ])('The map requires an output of "%s" at position "%s"', (output, position, input) => {
    expect(part1.sensorBoost(input).visible_asteroids).toEqual(output)
  })

  // Actual test, Part 1
  it('should produce the correct value for the input data', async () => {
    expect(part1.sensorBoost(await data).visible_asteroids).toMatchSnapshot()
  })
})

describe('Part 2', () => {
  it('Should find the best monitoring system, and get the 200th vaporized asteroid', async () => {
    const map = [
      '.#..##.###...#######',
      '##.############..##.',
      '.#.######.########.#',
      '.###.#######.####.#.',
      '#####.##.#.##.###.##',
      '..#####..#.#########',
      '####################',
      '#.####....###.#.#.##',
      '##.#################',
      '#####.##.###..####..',
      '..######..##.#######',
      '####.##.####...##..#',
      '.#####..#.######.###',
      '##...#.##########...',
      '#.##########.#######',
      '.####.#.###.###.#.##',
      '....##.##.###..#####',
      '.#.#.###########.###',
      '#.#.#.#####.####.###',
      '###.##.####.##.#..##',
    ].join('\n')

    // 802 because 8,2 => x * 100 + y
    expect(part2.findVaporizedAsteroid(map, 200)).toEqual(802)
  })

  // Actual test, Part 2
  it('should produce the correct value for the input data', async () => {
    expect(part2.findVaporizedAsteroid(await data, 200)).toMatchSnapshot()
  })
})
