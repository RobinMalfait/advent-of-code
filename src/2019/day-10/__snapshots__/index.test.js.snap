// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Part 1 should produce the correct value for the input data 1`] = `309`;

exports[`Part 2 should produce the correct value for the input data 1`] = `416`;
